text/0000-multiarch-builders-and-package.md

```diff
@@ -0,0 +1,1119 @@
+image = "docker.io/paketobuildpacks/build-jammy-tiny:0.2.3"
+
+[[targets]]
+os = "linux"
+arch = "amd64"
+
+[[targets]]
+os = "linux"
+arch = "arm64"
+
+# Stack (deprecated) 
+[stack]
+id = "io.buildpacks.stacks.jammy.tiny"
+# This image is used at build-time
+build-image = "docker.io/paketobuildpacks/build-jammy-tiny:0.2.3"
+# This image is used at runtime
+run-image = "index.docker.io/paketobuildpacks/run-jammy-tiny:latest"
+```
+
+As we can see, the proposal is based on the assumption that the `run-image`, `build-image` and `buildpacks` to include 
+in the builder are **multi-arch artifacts**, and we can reach them by reading an 
+[image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md)
+
+Packaging a multi-arch builder will require the output to be **publish** to a registry.
+
+```bash
+pack builder create my-jammy-builder --config ./builder.toml --publish
+```
+> **Important**
+> Similar to the `buildpack package,` pack will determine a multi-arch builder must be created based on the multiple targets defined.
+
+In this case `pack` will follow the builder creation process for **each provided target**, 
+pulling the correct (based on os/arch) buildpacks, build and run images and creating different builders images that are 
+exported and combined into an [image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md)
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Buildpack Package
+
+As a quick summary, our current process to create a buildpack package involves:
+
+- The end-users defined `os` for the OCI image using the [package.toml](https://buildpacks.io/docs/reference/config/package-config/).
+- The only values allowed are `linux` and `windows` and by default when is not present, `linux` is being used.
+- When exporting to daemon, the `docker.OSType` must be equal to `platform.os`
+
+### To keep compatibility 
+
+We propose:
+- Deprecate the `platform.os` field from [package.toml](https://buildpacks.io/docs/reference/config/package-config/). It will be removed after two pack releases with the new feature
+- When `platform.os` is present in [package.toml](https://buildpacks.io/docs/reference/config/package-config/), throw a warning messages indicating the field will be removed 
+and `--target` flag must be used
+- When `platform.os` is not present in [package.toml](https://buildpacks.io/docs/reference/config/package-config/) and `--target` flag is not used, throw a warning messages indicating 
+a new `--target` flag is available
+- Keep doing our current process to package a buildpack
+
+### To improve user experience
+
+We propose:
+- Add a new `--target` flag with the follwing format `[os][/arch][/variant]:[name@version]` to build for a particular target, once the `platform.os` field is removed, 
+this will be the way for end-users to specify the platform for which they want to create single OCI artifact.
```

Comment:
I like to think of `--target` as more of a helpful way to narrow down the os/arch when buildpack.toml has multiple definitions, or to avoid creating a manifest list. Is that correct?

Reply:
Yes, **exactly** when you actually want to avoid creating multiple images, then `target` will help you